### PR TITLE
fixes issue 239: classes derived from EnableQueryAttribute unable to unit test OnActionExecuted in isolation

### PR DESCRIPTION
**Pull request by howcheng**
see #239

### CODE REVIEW

File: src/Microsoft.AspNetCore.OData/Query/EnableQueryAttribute.cs [hunk 1]
```diff
@@ -542,7 +542,7 @@ private ODataQueryOptions CreateAndValidateQueryOptions(HttpRequest request, ODa
         {
             RequestQueryData requestQueryData = request.HttpContext.Items[nameof(RequestQueryData)] as RequestQueryData;
 
-            if (requestQueryData.QueryValidationRunBeforeActionExecution)
+            if (requestQueryData?.QueryValidationRunBeforeActionExecution ?? false)
```
Comment (gathogojr) [inline src/Microsoft.AspNetCore.OData/Query/EnableQueryAttribute.cs:545]: It took me a bit of time to understand this change. An easier to fix would be:
```suggestion
            if (requestQueryData != null && requestQueryData.QueryValidationRunBeforeActionExecution)
```

**Review by KenitoInc — APPROVED** [review]
Fix the change requested by @gathogojr and the PR will be good to merge

**Comment by KenitoInc** [comment]
@howcheng Kindly sign the CLA agreement

**Review by KenitoInc — APPROVED** [review]

**Review by gathogojr — APPROVED** [review]

**Comment by habbes** [comment]
Replaced by #776

**Comment by howcheng** [comment]
@KenitoInc Where can I find the CLA agreement? Thanks.

**Comment by KenitoInc** [comment]
> @KenitoInc Where can I find the CLA agreement? Thanks.

We copied your code to #776 
That's why we closed this PR

**Comment by howcheng** [comment]
@KenitoInc OK, so I assume I don't need to sign anything then. Thanks.

**Pull request closed without merging**